core/src/main/java/org/apache/carbondata/core/locks/CarbonLockFactory.java

```diff
@@ -52,44 +55,56 @@
    */
   public static ICarbonLock getCarbonLockObj(AbsoluteTableIdentifier absoluteTableIdentifier,
       String lockFile) {
-
-    String tablePath = absoluteTableIdentifier.getTablePath();
+    String tablePath;
+    if (lockPath.isEmpty()) {
+      tablePath = absoluteTableIdentifier.getTablePath();
+    } else {
+      if (absoluteTableIdentifier
+          .getCarbonTableIdentifier().getTableId().isEmpty()) {
+        throw new RuntimeException("Table id is empty");
+      }
+      tablePath = lockPath + CarbonCommonConstants.FILE_SEPARATOR + absoluteTableIdentifier
+          .getCarbonTableIdentifier().getTableId();
+    }
     if (lockTypeConfigured.equals(CarbonCommonConstants.CARBON_LOCK_TYPE_ZOOKEEPER)) {
-      return new ZooKeeperLocking(absoluteTableIdentifier, lockFile);
-    } else if (tablePath.startsWith(CarbonCommonConstants.S3A_PREFIX) ||
-        tablePath.startsWith(CarbonCommonConstants.S3N_PREFIX) ||
-            tablePath.startsWith(CarbonCommonConstants.S3_PREFIX)) {
+      return new ZooKeeperLocking(tablePath, lockFile);
+    } else if (tablePath.startsWith(CarbonCommonConstants.S3A_PREFIX) || tablePath
+        .startsWith(CarbonCommonConstants.S3N_PREFIX) || tablePath
+        .startsWith(CarbonCommonConstants.S3_PREFIX)) {
       lockTypeConfigured = CarbonCommonConstants.CARBON_LOCK_TYPE_S3;
-      return new S3FileLock(absoluteTableIdentifier, lockFile);
+      return new S3FileLock(tablePath,
```

Comment:
If table path is s3 then check for lockpath and locktype. if they are not configured throw exception.

Reply:
currently lock type is only used for Zookeeper locking other directly create the desired lock based on the path URI.